theano/tensor/nnet/tests/test_abstract_conv.py

```diff
@@ -1,5 +1,41 @@
+import numpy
 import unittest
+import itertools
+
+import theano
+from theano import tensor
+from theano.tests import unittest_tools as utt
+from theano.tensor.nnet import corr, abstract_conv as conv
 from theano.tensor.nnet.abstract_conv import get_conv_output_shape
+from theano.tensor.nnet.conv import ConvOp
+from theano.tensor.nnet.corr import (CorrMM, CorrMM_gradWeights,
+                                     CorrMM_gradInputs)
+from theano.tensor.nnet.ConvGrad3D import ConvGrad3D
+from theano.tensor.nnet.ConvTransp3D import ConvTransp3D
+
+
+def conv_corr(inputs, filters, border_mode="valid", subsample=(1, 1),
+              conv_mode='conv'):
+    if conv_mode == 'conv':
+        filters = filters[:, :, ::-1, ::-1]
+    return corr.CorrMM(border_mode, subsample)(inputs, filters)
```

Comment:
I think that the chance of all implementations getting their gradient wrong in the same way is very small.

And since we compare against all of them, this should ferret out if we ever break the gradient on CorrMM.

Also, just because CuDNN is a single op does not shield it from gradient errors...